### PR TITLE
Rewrite exercise to use league/route and symfony/string

### PR DESCRIPTION
**Pull request by AydinHassan**
Hey @mikeymike would appreciate a review here.

Basically all the deps we used in this exercise are not updated and do not work with PHP 8.1. So I had to search for a new routing library library. They are all not very simple, with `league/route` being the best one I could find. It is still more complicated, requiring you to marshal the request objects and output the responses, so you need to learn about PSR7 as well now and pull in an additional 2 libraries. 

`symfony/string` seemed like a good alternative for stringy but it doesn't support swapping case so I changed that to snake case.

Is the problem file understandable or too complicated?

### CODE REVIEW

File: exercises/dependency-heaven/problem/problem.md [hunk 1]
```diff
@@ -32,9 +32,61 @@ A request with `data = "you know you love it, don't you?"` should return the res
 }
 ```
 
-You should use the routing library `klein/klein` for this task pulling it in as a dependency through Composer. 
+You should use the routing library `league/route` for this task, pulling it in as a dependency through Composer. `league/route` allows us to define actions which respond to requests based on the request URI and HTTP method.
 
-You will also be required to use `danielstjules/stringy` to manipulate the data as this correctly handles multibyte characters. 
+The library works by accepting a PSR7 request and returns to you a PSR7 response. It is up to you to marshall the request object and output the response to the browser.
```
Comment (AydinHassan) [inline exercises/dependency-heaven/problem/problem.md]:
```suggestion
The library works by accepting a PSR7 request and returns to you a PSR7 response. It is up to you to marshal the request object and output the response to the browser.
```

File: exercises/dependency-heaven/problem/problem.md [hunk 2]
```diff
@@ -32,9 +32,61 @@ A request with `data = "you know you love it, don't you?"` should return the res
...
+
+There are a few other components we need, in order to use `league/route`:
+
+ * **laminas/laminas-diactoros** - For the PSR7 requests and responses.
+ * **laminas/laminas-httphandlerrunner** - For outputting the PSR7 response to the browser.
+
+`laminas/laminas-diactoros` is a PSR7 implementation. PSR's are standards defined by the PHP-FIG, a committee of PHP projects, attempting to increase interoperability in the PHP ecosystem. PSR7 is a standard for modelling HTTP requests. We can use the `laminas/laminas-diactoros` package to marshall a PSR7 request object from the PHP super globals like so:
```
Comment (AydinHassan) [inline exercises/dependency-heaven/problem/problem.md]:
```suggestion
`laminas/laminas-diactoros` is a PSR7 implementation. PSR's are standards defined by the PHP-FIG, a committee of PHP projects, attempting to increase interoperability in the PHP ecosystem. PSR7 is a standard for modelling HTTP requests. We can use the `laminas/laminas-diactoros` package to marshal a PSR7 request object from the PHP super globals like so:
```

File: exercises/dependency-heaven/problem/problem.md [hunk 3]
```diff
@@ -32,9 +32,61 @@ A request with `data = "you know you love it, don't you?"` should return the res
...
+
+```php
+(new \Laminas\HttpHandlerRunner\Emitter\SapiEmitter())->emit($response);
+```
+
+Where `$response` is a PSR7 response.
+
+In between this, you will need to define your routes and execute the dispatching mechanism to receive a response. Refer to the `league\route` [documentation](https://route.thephpleague.com/5.x/usage/).
+
+Each route action will be passed the PSR7 requestx§x where you can access the request parameters and body. To access the `data` key from the request body, you would use the following:
```
Comment (mikeymike) [inline exercises/dependency-heaven/problem/problem.md]: typo on request here

**Review by mikeymike — COMMENTED** [review]
Nice work on this, I think you've done a great job on the problem file, and while the solution is more complicated it's indicative of the real world and prompts dependency documentation comprehension 😄

File: exercises/dependency-heaven/problem/problem.md (hunk 3 above)
Comment (AydinHassan) [inline exercises/dependency-heaven/problem/problem.md]:
```suggestion
Each route action will be passed the PSR7 request where you can access the request parameters and body. To access the `data` key from the request body, you would use the following:
```
```suggestion
Each route action will be passed the PSR7 requestx§x where you can access the request parameters and body. To access the `data` key from the request body, you would use the following:
```

**Pull request merged**